### PR TITLE
PWA

### PR DESCRIPTION
**Pull request by Ghostbird**
Make front-end an installable PWA:

- Add manifest
- Add serviceworker with caching
- Add necessary metadata
- Add icons and splashscreens (placeholders)
- Disable Vue HTTPS server in development mode because PWAs are not installable on sites with untrusted certificates.
- Doesn't work in Firefox yet!

## Screenshots

Install icon in Chromium
Installable

Install prompt in Chromium
Install prompt

Installed and running in standalone Chromium
Installed

Integration with operation system start menu (Debian / Cinnamon)
OS integration

### CODE REVIEW

File: frontend/public/manifest.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
+{
+  "$schema": "https://json.schemastore.org/web-manifest",
+  "name": "Universal BUK Games",
+  "short_name": "UBG",
+  "description": "The Universal BUK Games on your device",
+  "lang": "en",
+  "start_url": ".",
+  "scope": "/",
+  "display": "fullscreen",
```
Comment (ruudkobes) [inline frontend/public/manifest.json:9]: standalone (the default) might be better, running as it's own app but allow notification bar etc to still show.

File: frontend/public/manifest.json
Comment (ruudkobes) [inline frontend/public/manifest.json:1]: Don't know if it matters much, but apparently the more formal extension is .webmanifest. I suppose as long as it is specified it should work, but we should verify if this is fine on ios/safari

**Review by ruudkobes — COMMENTED** [review]
I think we should have a look at the splashsscreen sizes when getting the official ones in

File: frontend/public/manifest.json (hunk 1 above)
Comment (Ghostbird) [reply frontend/public/manifest.json:9]: I based my choice on the [MDN `display` documentation](https://developer.mozilla.org/en-US/docs/Web/Manifest/display). Fullscreen should refer here to the display of the PWA within the browser, not the PWA in the OS.

> `fullscreen` All of the available display area is used and no user agent [chrome](https://developer.mozilla.org/en-US/docs/Glossary/Chrome) is shown.

I based it on the fact that they explicitly mention the _chrome_ (*The* chrome explicitly refers to Browser UI elements), and the fact that they state reference _all of the **available** display area_ made me think that `fullscreen` does not hide the OS title bar, and does not mean a full-screen native application.

This [this stackoverflow answer](https://stackoverflow.com/questions/33129326/chrome-web-app-manifest-differences-between-display-types) however, states that `standalone` is what we desire, and `fullscreen` hides the status bar.

Note that this is _not true_ on my desktop. There a PWA installed with `fullscreen` mode behaves as expected. It is not fullscreen, and there's (almost) no visible chrome.

Comment (ruudkobes) [reply frontend/public/manifest.json:9]: Phone is definitely the place to test. Fullscreen is also ignored on my desktop, but that might be an adjustment on the desktop side of things. I think standalone is the best match for now.

File: frontend/public/manifest.json
Comment (Ghostbird) [reply frontend/public/manifest.json:1]: Turns out the content type `application/manifest+json` is automatically set correctly. So I changed it.

File: frontend/public/manifest.json (hunk 1 above)
Comment (Ghostbird) [reply frontend/public/manifest.json:9]: Standalone seems to work best indeed.

**Comment by Ghostbird** [comment]
About the splash screens. The sizes are some common Apple device screen sizes, maybe a bit outdated though.
I haven't managed to get the PWA icon working on Firefox. I checked against manifests of both https://aksjon.bcc.no and mozilla's PWA examples, and I can't figure out what I'm doing wrong. So for now I'm going to assume it's either a caching issue, or caused because I'm in a development set-up. Chromium also needed some hoops before PWAs worked _at all_ on https://localhost:8080.

**Pull request merged**